test/cases/testdata/comparisonexpr/test-comparisonexpr-0620.yaml

```diff
@@ -0,0 +1,17 @@
+cases:
+# Leaving out floats as map keys until this is resolved:
+# https://github.com/open-policy-agent/opa/issues/4797
+- modules:
+  - |
+    package comparison
+
+    p {
+      1 == 1.0
+      [1] == [1.0]
+      {1: 1} == {1: 1.0}
+      {"x": [1, 2, {"b": 3.0}]} == {"x": [1, 2, {"b": 3}]}
+    }
+  note: 'comparisonexpr/numbers: ints and floats nested comparison'
```

Comment:
if this fails as a package, it'll be harder to debug than if these were separate test cases 🤔

Reply:
Fair enough, I'll make it so 👍